lib/acl-checker.js

```diff
@@ -144,6 +146,49 @@ class ACLChecker {
   isAcl (resource) {
     return resource.endsWith(this.suffix)
   }
+
```

Comment:
Is this refactoring necessary to solve implement this issue? I can't tell. If not, would it be better to do the refactor in a separate PR to ease the review of this feature?

Reply:
If I don't do it like this, I will copy the code instead of refactoring it =/ I thought this a better approach.

That said, I agree it's a bit messy, and am open for suggestions on other approaches.